tests/framework/behaviors/TimestampBehaviorTest.php

```diff
@@ -188,6 +188,41 @@ public function testUpdateRecordExpression()
         $this->assertEquals($enforcedTime, $model->created_at, 'Create time has been set on update!');
         $this->assertEquals(date('Y'), $model->updated_at);
     }
+    
+    public function testTouchIsNewRecordGenerateException()
+    {
+        ActiveRecordTimestamp::$behaviors = [
+            'timestamp' => [
+                'class' => TimestampBehavior::className(),
+                'value' => new Expression("strftime('%Y')"),
+            ],
+        ];        
+        $model = new ActiveRecordTimestamp();
+        
+        $this->setExpectedException('yii\base\InvalidCallException');
```

Comment:
https://github.com/yiisoft/yii2/blob/master/framework/behaviors/TimestampBehavior.php#L129

the next line tries to update an attribute from a new record using the `touch()` method